File: app/Http/Controllers/SpecialistPatterns/LR/LR1SP/graph/AbstractGOFactory.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\LR\LR1SP\graph;

// Абстрактная фабрика для создания графических объектов
abstract class AbstractGOFactory {

    // Метод для создания точки
    public abstract function createPoint(): Point;

    // Метод для создания окружности
    public abstract function createCircle(): Circle;
}

File: app/Http/Controllers/SpecialistPatternsJava/Lab1/Lab1Java/src/graph/AbstractGOFactory.java
package graph;

// Абстрактный класс AbstractGOFactory представляет собой фабрику для создания графических объектов.
public abstract class AbstractGOFactory {
    // Абстрактный метод для создания объекта класса Point.
    public abstract Point createPoint();

    // Абстрактный метод для создания объекта класса Circle.
    public abstract Circle createCircle();
}
